XLS-0056d-batch/README.md

```diff
@@ -0,0 +1,615 @@
+|`TransactionHash`|✔️|`string`|`STUInt256`|
+|`TransactionResult`|✔️|`string`|`STBlob`|
+
+Some important things to note:
+* It is possible that all transactions will not be included in this list. For example, when using the `ONLYONE` mode, if the first transaction succeeds, then the rest of the transactions will not even be processed. 
+* Transactions will only be included in the ledger if their result code is `tesSUCCESS` _and_ if the outer transaction has a result code of `tesSUCCESS`. For example, if the `ALLORNOTHING` mode is used and _any_ inner transaction fails, _none_ of the inner transactions will be included in the ledger.
+
+#### 2.6.2. Inner Transactions
+
+Each inner transaction will contain the metadata for its own processing. Only the inner transactions that were actually committed to the ledger will be included. This makes it easier for legacy systems to still be able to process `Batch` transactions as if they were normal.
+
+There will also be a pointer back to the parent outer transaction (`parent_batch`), for ease of development (similar to the `nftoken_id` field).
+
+## 3. Transaction Common Fields
+
+This standard adds one new field to the [transaction common fields](https://xrpl.org/docs/references/protocol/transactions/common-fields/):
+
+| Field Name | Required? | JSON Type | Internal Type |
+|------------|-----------|-----------|---------------|
+|`BatchTxn`| |`object`|`STObject`|
+
+### 3.1. `BatchTxn`
+
+The `BatchTxn` inner object **must** be included in any inner transaction of a `Batch` transaction (and must not be included in any other transactions). Its inclusion:
+* Prevents hash collisions between identical transactions (since sequence numbers aren't included).
+* Ensures that every transaction has a sequence number associated with it, so that created ledger objects that use it in their ID generation can still operate.
+* Allows users to more easily organize their transactions in the correct order.
+
+The fields contained in this object are:
+
+| Field Name | Required? | JSON Type | Internal Type |
+|------------|-----------|-----------|---------------|
+|`OuterAccount`|✔️|`string`|`AccountID`|
+|`Sequence`| |`number`|`UInt32`|
+|`TicketSequence`| |`number`|`UInt32`|
+|`BatchIndex`|✔️|`number`|`UInt8`|
+
+#### 3.1.1. `OuterAccount`
+
+This is the account that is submitting the outer `Batch` transaction.
+
+#### 3.1.2. `Sequence`
+
+This is the next valid sequence number of the inner transaction sender (i.e. all inner _and_ outer transactions from the same account will have the same `BatchTxn.Sequence` value). Its inclusion ensures that there are no hash collisions with other `Batch` transactions, and also allows for proper sequence number calculation (see section 4.2 for more details).
+
+In other words, a single-account `Batch` transaction will have the same sequence number in the outer transaction _and_ all the inner transactions.
+
+#### 3.1.3. `TicketSequence`
```

Comment:
Need to clarify something here. We add 1 on inner txn where outer.account == inner.account because we assume the outer txn consumed a sequence. If we use a ticket on the outer txn then we cannot add one but we wont know that based on the data from `sfBatchTxn`.

Why dont we just pre calculate the sequence? If the failure is anything but tes or tec then the txn is reverted regardless of the atomic type. If its a tes or tec it will consume the sequence. So there's not much value in calculating it in the `getSeqProxy` idk.

Reply:
What do you mean by `pre-calculate the sequence`?

Reply:
Ignore that comment